docs/architecture/adr-052-tendermint-mode.md

```diff
@@ -0,0 +1,73 @@
+# ADR 052: Tendermint Mode
+
+## Changelog
+
+* 27-11-2019: Initial draft from ADR-051
+* 13-01-2020: Separate ADR Tendermint Mode from ADR-051
+
+## Context
+
+- Fullnode mode : fullnode mode does not have any capability to participate on consensus
+- Validator mode : this mode is exactly same as existing state machine behavior. sync without voting on consensus, and participate consensus when fully synced
+- Seed mode : lightweight seed mode maintaining an address book, p2p like [TenderSeed](https://gitlab.com/polychainlabs/tenderseed)
+
+## Decision
+
+We would like to suggest a simple Tendermint mode abstraction
+
+- Which reactor, component to include for each node
+    - fullnode *(default)*
+        - switch, transport
+        - all reactors
+        - mempool, state
+        - rpc
+        - *~~no privValidator(priv_validator_key.json, priv_validator_state.json)~~*
+    - validator
+        - switch, transport
+        - all reactors
+        - mempool, state
+        - rpc
+        - with privValidator(priv_validator_key.json, priv_validator_state.json)
+    - seed
+        - switch, transport
+        - only pexReactor
+- Configuration, cli command
+    - We would like to suggest by introducing `mode` parameter in `config.toml` and cli
+    - `mode = "{{ .BaseConfig.Mode }}"` in `config.toml`
+    - `tendermint node --mode validator`  in cli
+    - fullnode | validator | seed (default: "fullnode")
+- RPC modification
+    - `host:26657/status`
+        - return empty `validator_info` when fullnode mode
+        - add return `mode` of this node
```

Comment:
this issue may have some thoughts on introducing which mode the node is in. https://github.com/tendermint/tendermint/issues/2817

Reply:
how about adding only line `return empty validator_info when fullnode mode` without line `add return mode of this node`?

Reply:
Yea this is starting to play into a redesign of the rpc (possibly having different levels, that is out of scope) so for now I'd say that is fine.